lib/constructs/callback-interface.js

```diff
@@ -0,0 +1,232 @@
+"use strict";
+
+const utils = require("../utils.js");
+const Types = require("../types.js");
+const Constant = require("./constant.js");
+
+class CallbackInterface {
+  constructor(ctx, idl) {
+    this.ctx = ctx;
+    this.idl = idl;
+    this.name = idl.name;
+    this.str = null;
+
+    this.requires = new utils.RequiresMap(ctx);
+
+    this.operation = null;
+    this.constants = new Map();
+
+    this._analyzed = false;
+    this._outputStaticProperties = new Map();
+  }
+
+  _analyzeMembers() {
+    for (const member of this.idl.members) {
+      switch (member.type) {
+        case "operation":
+          if (this.operation !== null) {
+            throw new Error(
+              `Callback interface ${this.name} has more than one operation`
```

Comment:
Nothing seems to prevent callback interfaces from having more than one operation. See e.g., this test case here: https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/bindings/tests/idls/core/test_callback_interface.idl;l=30;drc=0aee4434a4dba42a42abaea9bfbc0cd196a63bc1. But on the other hand, I can't actually find a spec that puts more than two operations into the same callback interface. Hmm.

We should probably file a spec bug.

Reply:
Wait, never mind.

> Callback interfaces must define exactly one regular operation.